docs/source/sparse.rst

```diff
@@ -1,146 +1,382 @@
+strided tensors is likely the most efficient approach.
 
-    This API is in beta and may change in the near future.
+.. _sparse-fill-value-docs:
 
-Torch supports sparse tensors in COO(rdinate) format, which can
-efficiently store and process tensors for which the majority of elements
-are zeros.
+Fill value
+++++++++++
 
-A sparse tensor is represented as a pair of dense tensors: a tensor
-of values and a 2D tensor of indices.  A sparse tensor can be constructed
-by providing these two tensors, as well as the size of the sparse tensor
-(which cannot be inferred from these tensors!)  Suppose we want to define
-a sparse tensor with the entry 3 at location (0, 2), entry 4 at
-location (1, 0), and entry 5 at location (1, 2).  We would then write:
+In Torch, the fill value is not specified explicitly but is implicitly
+interpreted by sparse algorithms. The implicit interpretation may vary
+in-between algorithms of different fields and applications. The most
+commonly assumed fill value is zero and it is assumed for all tools
+that are available via :mod:`torch` namespace. When some sparse tensor
+functionality assumes a non-zero or indefinite fill value, the
+corresponding tools are provided via :mod:`torch.sparse`
+namespace. Please see the documentation strings of the corresponding
+functions for a particular definition of the fill value.
 
-    >>> i = torch.LongTensor([[0, 1, 1],
-                              [2, 0, 2]])
-    >>> v = torch.FloatTensor([3, 4, 5])
-    >>> torch.sparse.FloatTensor(i, v, torch.Size([2,3])).to_dense()
-     0  0  3
-     4  0  5
-    [torch.FloatTensor of size 2x3]
+.. _sparse-hybrid-docs:
 
-Note that the input to LongTensor is NOT a list of index tuples.  If you want
+Hybrid tensors
+++++++++++++++
+
+Torch supports hybrid sparse tensors with dimensions split into sparse
+dimensions and dense dimensions. Dense dimensions always follow sparse
+dimensions, that is, mixing of dense and sparse dimensions is not
+supported. One can interpret hybrid tensors as sparse tensors with
+elements being (contiguous) strided tensors. We use (M +
+K)-dimensional tensor to denote a N-dimensional hybrid sparse tensor
+where M and K are the number of sparse and dense dimensions,
+respectively, such that M + K == N holds.
+
+See :ref:`Hybrid COO tensors <sparse-hybrid-coo-docs>`.
+
+Sparse tensor formats
++++++++++++++++++++++
+
+Torch supported sparse storage formats are:
+
+- COO(rdinate) format where tensor elements are stored using the lists
+  of indices and the corresponding values. Unspecified elements have
+  values equal to the fill value. To construct a sparse tensor in COO
+  format, use :func:`torch.sparse_coo_tensor` function.
+
+.. _sparse-coo-docs:
+
+Sparse COO tensors
+******************
+
+The sparse COO tensor is represented as a pair of dense tensors: a 2D
+tensor of indices and a tensor of values. A sparse COO tensor can be
+constructed by providing these two tensors, as well as the size of the
```

Comment:
I would separate the discussion of constructing a sparse tensor to it's own section, and let the previous section focus on the mathematics. Maybe even mentioning how PyTorch represents the strided tensor is too much there.